src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java

```diff
@@ -324,60 +326,60 @@ private IAIState findTrees()
      */
     private IAIState findTree()
     {
-        final IBuilding building = getOwnBuilding();
+        final BuildingLumberjack building = getOwnBuilding();
         worker.getCitizenData().setVisibleStatus(SEARCH);
 
-        if (pathResult == null || pathResult.treeLocation == null)
+        if (pathResult != null && pathResult.isComputing())
         {
-
-            final BuildingLumberjack buildingLumberjack = (BuildingLumberjack) building;
-            final Map<String, List<ItemStorage>> copy = buildingLumberjack.getCopyOfAllowedItems();
-            if (buildingLumberjack.shouldRestrict())
+            return getState();
+        }
+        if (pathResult == null)
+        {
+            final Map<String, List<ItemStorage>> copy = building.getCopyOfAllowedItems();
+            if (building.shouldRestrict())
             {
-                final BlockPos startPos = buildingLumberjack.getStartRestriction();
-                final BlockPos endPos = buildingLumberjack.getEndRestriction();
-
-                pathResult = worker.getNavigator().moveToTree(
-                  startPos, endPos,
-                  1.0D,
-                  copy.getOrDefault(SAPLINGS_LIST, Collections.emptyList()),
-                  worker.getCitizenColonyHandler().getColony()
-                );
+                final BlockPos startPos = building.getStartRestriction();
+                final BlockPos endPos = building.getEndRestriction();
+
+                pathResult = worker.getNavigator()
+                    .moveToTree(startPos,
+                        endPos,
+                        1.0D,
+                        copy.getOrDefault(SAPLINGS_LIST, Collections.emptyList()),
+                        worker.getCitizenColonyHandler().getColony());
             }
             else
             {
                 pathResult = worker.getNavigator()
-                               .moveToTree(SEARCH_RANGE + searchIncrement,
-                                 1.0D,
-                                 copy.getOrDefault(SAPLINGS_LIST, Collections.emptyList()),
-                                 worker.getCitizenColonyHandler().getColony());
+                    .moveToTree(SEARCH_RANGE + searchIncrement,
+                        1.0D,
+                        copy.getOrDefault(SAPLINGS_LIST, Collections.emptyList()),
+                        worker.getCitizenColonyHandler().getColony());
             }
+            return getState();
         }
         if (pathResult.isPathReachingDestination())
         {
-            return setNewTree();
-        }
-        if (pathResult.isCancelled())
```

Comment:
I have really no idea what that means here - more like why if path is cancelled we switched to gathering

Reply:
Good question, are there still other ways to go into GATHERING?